core/templates/dev/head/pages/library/Library.js

```diff
@@ -40,34 +40,56 @@ oppia.controller('Library', [
     // Keeps track of the index of the left-most visible card of each group.
     $scope.leftmostCardIndices = [];
 
-    $http.get('/libraryindexhandler').success(function(data) {
-      $scope.libraryGroups = data.activity_summary_dicts_by_category;
-
-      $rootScope.$broadcast(
-        'preferredLanguageCodesLoaded', data.preferred_language_codes);
-
-      $rootScope.loadingMessage = '';
-
-      // Pause is necessary to ensure all elements have loaded, same for
-      // initCarousels.
-      // TODO(sll): On small screens, the tiles do not have a defined width.
-      // The use of 214 here is a hack, and the underlying problem of the tiles
-      // not having a defined width on small screens needs to be fixed.
-      $timeout(function() {
-        tileDisplayWidth = $('exploration-summary-tile').width() || 214;
-      }, 20);
-
-      // Initialize the carousel(s) on the library index page.
-      $timeout(initCarousels, 390);
-
-      // The following initializes the tracker to have all
-      // elements flush left.
-      // Transforms the group names into translation ids
-      $scope.leftmostCardIndices = [];
-      for (i = 0; i < $scope.libraryGroups.length; i++) {
-        $scope.leftmostCardIndices.push(0);
-      }
-    });
+    $scope.inRankMode = ($window.location.pathname.indexOf(
```

Comment:
Modified the if condition to:
```
$scope.inGroupMode = ($window.location.pathname.indexOf(
      'recently_published') === 9) || ($window.location.pathname.indexOf(
      'top_rated') === 9);
```
which holds for only recently published and top rated pages.